File: tools/kystrap/templates/python/server/start.py
from grpclib.server import Server
from grpclib.utils import graceful_exit

from server.server import {{ .name | ToPascal }}Server


async def serve(*, host='0.0.0.0', port=50051):
    server = Server([{{ .name | ToPascal }}Server()])

    # Note: graceful_exit isn't supported in Windows
    with graceful_exit([server]):
        await server.start(host, port)
        print(f'🌐 {{ .name | ToPascal }}Server is running on http://{host}:{port}\nPress Ctrl + C to exit.')
        await server.wait_closed()

File: integrations/tendermint-go/server/start.go
package server

import (
	"fmt"
	pb "github.com/KYVENetwork/kyvejs/integrations/tendermint-go/proto/kyverdk/runtime/v1"
	"google.golang.org/grpc"
	"log"
	"net"
)

const (
	host = "0.0.0.0"
	port = "50051"
)

func StartServer() {
	// Initialize the gRPC server and listen on port 50051
	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%s", host, port))
	if err != nil {
		log.Fatalf("Failed to listen on port 50051: %v", err)
	}

	// Create a new gRPC server instance
	server := grpc.NewServer()

	// Register the Tendermint service with the gRPC server
	pb.RegisterRuntimeServiceServer(server, &TendermintGoServer{})

	// Start serving incoming connections
	fmt.Printf(fmt.Sprintf("🌐 TendermintGoServer is running on on http://%s:%s...\nPress Ctrl + C to exit.\n", host, port))
	err = server.Serve(listener)
	if err != nil {
		log.Fatalf("Failed to serve gRPC server: %v", err)
	}
}
